Racing team full PR

alo-20 opened the pull request:
First PR for Racing team. Includes:
- More set paths for testing control
- Cone detection logic
- Launch Control logic
- Racing Planner
- Stanley controller
- velocity_profile which generates desired velocities for points along a racing path
- trajectory.py
- A few extra tools for plotting, generating paths, and testing

diegoasfmravpp committed:
gg diagram for dynamic limits

diegoasfmravpp committed:
speed and accel plots

diegoasfmravpp committed:
50m route and launch file

diegoasfmravpp committed:
braking at route end for pp

Demon16th committed:
launch control added

diegoasfmravpp committed:
plotting changes

alo-20 committed:
Add 65m and right turn routes

Demon16th committed:
add better right turn paths

diegoasfmravpp committed:
phone accel data

alo-20 committed:
add methods in trajectory.py to calculate upcoming turn radius

Demon16th committed:
add logic in pure_pursuit to set speed based on curve radius. Change curve radius calculation in trajectory.py. Add generate_circle.py to generate circle paths and add circle paths in knowledge/routes.

diegoasfmravpp committed:
gg data switched to json

Demon16th committed:
add figure8 path generator and figure8 path

Demon16th committed:
modify fit_curve_radius in trajectory.py to use the min (tighest) turn radius of the next 3 segments of 3 points

Demon16th committed:
update generate figure8 and figure8 path

Commit:
gg logs and launch control tests

Demon16th committed:
pp speeds

diegoasfmravpp committed:
fixed gg plot

diegoasfmravpp committed:
velocity profile based on existing route

laneccolin committed:
Add files via upload

laneccolin committed:
Update fixed_route.yaml

laneccolin committed:
Update fixed_route.yaml

laneccolin committed:
Update current.yaml

Adi9103 committed:
changed launch ctrl to gem

ShellyRiver committed:
initial trajectory generation and feasibility check functions

htliang517 committed:
Update waypoint generation and combine with trajectory planning

diegoasfmravpp committed:
integrating  velocity profile into stanley

ShellyRiver committed:
trajectory generation can take x in state and speed is not fixed (so total travel distance can vary)

Demon16th committed:
updated stanley + racing velocities, working sim

Demon16th committed:
lower speed for high cte in racing

ShellyRiver committed:
try to run in sim

Adi9103 committed:
moved logic

ShellyRiver committed:
try to run in simulation

Adi9103 committed:
updated launch_ctrl, added pure_pursuit

htliang517 committed:
Update & fix simulation problem for planner

htliang517 committed:
Fix U-turn scenario

Jugthegreat committed:
Update current.yaml

updated working paramters

Demon16th committed:
move launch control logic out of gem.py and gem_hardware.py and into the planners (pure_pursuit and stanley)

Demon16th committed:
fix typo in current.yaml

ShellyRiver committed:
add 90turn case and test code

ShellyRiver committed:
refine 4-cone course planned trajectory

diegoasfmravpp committed:
visualization changes for stanley logs

htliang517 committed:
Fixed code to work with simulator

dh-agarwal committed:
CONE DETECTOR DUMP

ShellyRiver committed:
to test velocity profile to get total time for unit test

ShellyRiver committed:
add velocity profiling to racing planner

ShellyRiver committed:
generate trajectory for whole slalom scenario

htliang517 committed:
Update planner to run both onboard and simulation

alo-20 committed:
add velocity profile plotting from behavior.json

diegoasfmravpp committed:
splines for path smoothing and updated stanley

htliang517 committed:
Update planner logic with real-time detection

dh-agarwal committed:
Update cone_detection.py

alo-20 committed:
remove commented out launch control code in gem_hardware.py

diegoasfmravpp committed:
added turn rate and accel limits to planner in
trajectory_generation_dynamics

dh-agarwal committed:
cone logging logic?

Adi9103 committed:
modularized launch_ctrl

alo-20 committed:
Remove logs and unneeded files

alo-20 committed:
Remove old versions of racing planner

alo-20 committed:
Add up-to-date version of racing planning

alo-20 committed:
add better version of velocity mapping accounting for steering rate

alo-20 committed:
Fix launch_control.py reset bug, remove launch control enable from current.yaml and move into launch file arg for stanley and pure_pursuit

alo-20 committed:
Clean up pure_pursuit racing logic

diegoasfmravpp committed:
velocity_profile.py cleanup

diegoasfmravpp committed:
velocity_profile.py cleanup

krishauser reviewed (changes requested):
Let's get the perception PR merged. The cones will no longer be Agents, but rather Obstacles.

diegoasfmravpp committed:
fixed route to trajectory with velocity to  work
by launch file without manual changes to  stanley

alo-20 commented on GEMstack/onboard/interface/gem.py:
We originally had some changes for launch control but moved the logic elsewhere. I can remove these changes.

alo-20 committed:
remove whitespace changes in gem_hardware

alo-20 committed:
Remove whitespace and unneeded changes in gem.py

alo-20 committed:
update stanley

alo-20 committed:
Fix launch control reset behavior and update trajectory.py

alo-20 committed:
remove launch control reset

alo-20 committed:
Add dynamics changes from other racing branches

alo-20 committed:
revert stanley

KenC1014 committed:
launch control logs

alo-20 committed:
Update search area for closest point

alo-20 committed:
Remove logs from pr branch

alo-20 committed:
Merge branch 's2025' into racing_pr_prep

alo-20 committed:
Re enable launch control reset and add comments

alo-20 committed:
update gem_simulator and racing_planning with latest

alo-20 committed:
Update launch files

alo-20 commented:
This should be ready for re-review. All changes that refer to Agents should just be for the Klampt simulation.

diegoasfmravpp committed:
duplicated code segment

krishauser commented on GEMstack/knowledge/defaults/current.yaml:
Don’t commit formatting changes

krishauser replied:
Are these parameter changes coordinated with the control team / parking team?

alo-20 replied:
We agreed with the control team to separate the racing tuning parameters. So this should no longer affect them. But the Baseline stanley parameters are the same that the control team has.

krishauser commented on GEMstack/knowledge/vehicle/dynamics.py:
Stray print statements, no real change

krishauser commented on GEMstack/knowledge/vehicle/gem_e4_dynamics.yaml:
Stray added comments?

alo-20 replied:
Removed extra comments

krishauser replied:
These accelerator / braking ranges do not appear coordinated with the control team's changes. @Jugthegreat did you agree to these changes?

krishauser commented on GEMstack/mathutils/transforms.py:
Don’t commit unnecessary changes

krishauser commented on GEMstack/onboard/interface/gem_simulator.py:
Do not hard code gear.

krishauser commented on GEMstack/state/trajectory.py:
Don’t add velocities to trajectory. They should be apparent when evaluating the derivative

alo-20 replied:
Our Velocity Profile calculates the fastest velocity we should be able to drive at given points along a path. The problem is if we just use times and we don't exactly meet the times (overshoot or something), then the car will try to drive faster than it should in order "catch up." Specifying velocities along the path prevents this issue.

krishauser replied:
It's not the function of a trajectory to specify velocity limits. That's more of the function of a route or a plan. We talked about the definition of a trajectory in class.

alo-20 replied:
They're basically just the desired velocities along the path. The control is not accurate enough to use timings alone. 
The velocity profile is calculated as part of the planning process, and then added to the path.

alo-20 replied:
The reason I said "fastest velocity" before is because since we're doing 'racing' we want to be going as fast as possible at any given point. But they're just desired speeds.

alo-20 replied:
We have now removed the velocities from the Trajectory. As you said after the presentations using the timings gives you the velocities. Since the times we look at are the expected times then it doesn't matter if you lag behind, it will still work.

krishauser commented on launch/planner_sim.yaml:
Rename to be more precise about what is being launched.

krishauser reviewed (changes requested):
The PR includes many stray commits. Also, a trajectory should not include a separate velocities attribute. Its velocities are determined by timing.

alo-20 committed:
Remove changes to dyanamics.py

alo-20 committed:
Remove changes from dynamics.py correctly

alo-20 committed:
Remove unncessary changes per PR comments

alo-20 committed:
Turn off velocity profile plotting by default

alo-20 committed:
remove formatting change from current.yaml

alo-20 committed:
remove whitespace

alo-20 committed:
remove extra comments from gem_e4_dyanamics

alo-20 committed:
Update launch files

ShellyRiver committed:
remove route in launch file

htliang517 committed:
Clean up racing planner code

alo-20 commented on GEMstack/onboard/interface/gem_simulator.py:
Agents are left here just for Sim usage, otherwise the sim won't work.

krishauser commented on GEMstack/onboard/perception/cone_detection.py:
Please take this out of the PR if nothing is changing

alo-20 replied:
My bad, I thought this was removed

krishauser reviewed (changes requested):
Commits not yet coordinated well with other teams.

alo-20 committed:
remove changes to cone_detection.py

alo-20 committed:
remove changes to gem_e4_dyanmics

alo-20 committed:
Separate racing tuning parameters in current.yaml for stanley and pure_pursuite

diegoasfmravpp committed:
changing desired speed for PID back to derivative eval and changing max speed back to 10.0

alo-20 committed:
Merge branch 's2025' into racing_pr_prep

alo-20 commented:
Changed current.yaml to have separate racing parameters as discussed with the control team. 
Removed velocities from Trajectory as discussed.

krishauser reviewed (changes requested):
The changes recently made didn't address the issues. Sorry, your PR wasn't successfully approved.

alo-20 commented:
I'm sorry, but what issues are still remaining? We addressed the Trajectory issue by taking out velocities, and we switched the racing settings in current.yaml to be separate from the other settings. I'm not sure what other issues there were, other than having a few "Agent" references left... but I mentioned in the PR comments that that was only so that we could still use the Klampt simulation. If you'd like we could take that out as well.

alo-20 committed:
add changes from control group

krishauser commented:
You made surface-level changes to address my review but you didn't test the code.

alo-20 committed:
Manual integration of control team logic

alo-20 committed:
Merge branch 's2025' into racing_pr_prep

alo-20 commented:
I did test the code in simulation, would you like to see it working?

alo-20 commented:
> I did test the code in simulation, would you like to see it working?

The reason why we did switch back to just timings is because Diego and I tested it last night and it worked in sim.

htliang517 committed:
Fix minor issue with unused reference test code indicated by autocheck.

htliang517 committed:
Fix warning of velocity_profile

alo-20 committed:
Remove cone agents and scenes for simulation

alo-20 committed:
remove racing_fixed_slalom
